spring-rabbit-test/src/main/java/org/springframework/amqp/rabbit/test/junit/SpringRabbitTest.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.amqp.rabbit.test.junit;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import org.springframework.amqp.rabbit.junit.RabbitAvailable;
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.core.annotation.AliasFor;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+
+/**
+ * Adds infrastructure beans to a Spring test context; do not use with Spring Boot since
+ * it has its own auto configuration mechanism.
+ *
+ * @author Gary Russell
+ * @since 2.3
+ *
+ */
+@SpringJUnitConfig
```

Comment:
`@SpringIntegrationTest` doesn't do this because you can't guarantee that end-user always going to use JUnit 5.
Also I'm not sure why have you made it based on the JUnit 5 `Condition`, nut not `ContextCustomizer` and `ContextCustomizerFactory` from Spring Test...

Reply:
Ah, right; makes sense.

I was thinking that it's not unreasonable to require JUnit5 starting with this version. But I will rework it.